### ISSUE TITLE
claim() and mainsnak() expect "snak()" as parameter

### ISSUE THREAD

**Issue by MHuberFaust** [UB-Mannheim/RaiseWikibase#14]
Me again :)

I worked through the examples and found this:

claim() and mainsnak() expect "snak()" as parameter, but snak() returns a dictionary.

I adjusted the parameters to accept a dictionary to fix it on my end.

Please let me know wether you change the code or the Readme.

Best,
Michael

PS: Thanks for this library, it helps a lot!

**Comment by renat-kaufmann** [comment]
I am not sure whether I understood you correctly.

[Claim() function](https://github.com/UB-Mannheim/RaiseWikibase/blob/main/RaiseWikibase/datamodel.py#L175) has the following arguments:
```python
def claim(prop='', mainsnak=snak(), qualifiers=[], references=[]):
```

[Mainsnak() function](https://github.com/UB-Mannheim/RaiseWikibase/blob/main/RaiseWikibase/datamodel.py#L198) has these arguments:
```python
def mainsnak(prop='', snak=snak(), qualifiers=[], references=[]):
```

So mainsnak()-function has parameter snak with default value equal to the returned value of the snak()-function which is indeed a dictionary. See [snak()-function](https://github.com/UB-Mannheim/RaiseWikibase/blob/main/RaiseWikibase/datamodel.py#L48-L172).

Could you share your example to clarify what you mean?

**Comment by MHuberFaust** [comment]
Sorry, it is indeed confusing.

Without trying to reproduce the error message (I will do it, when the following does not help). It said something like "dictionary is not callable".

claim() and mainsnak() did not accept a dictionary as input for the argument "mainsnak"/"snak". I changed:

`
def mainsnak(prop='', snak=snak(), qualifiers=[], references=[]):
`

to:


`def mainsnak(prop='', snak={}, qualifiers=[], references=[]):`


and it worked fine. I don not know the inner workings of python enough to explain why, though. 

I ran it via a jupyter notebook, don't know if that is relevant.
